tests/unit_tests/model_utils/test_reservoir_shuffle.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from exasol_data_science_utils_python.model_utils.reservoir_shuffle import ReservoirShuffle
+
+
+class TestValue:
+    def __init__(self, value):
+        self.value = value
+
+    def __eq__(self, other):
+        return other.value == self.value
+
+    def __gt__(self, other):
+        return other.value > self.value
+
+    def __repr__(self):
+        return f"TestValue({self.value})"
+
+map_to_test_value = np.vectorize(lambda y: TestValue(y))
+
+@pytest.mark.parametrize("input_size,batch_size,buffer_size,columns_count,column_generator", [
+    (50, 10, 20, 1, lambda i, x: x),
+    (50, 10, 20, 2, lambda i, x: x),
+    (50, 20, 20, 1, lambda i, x: x),
+    (50, 10, 25, 1, lambda i, x: x),
+    (100, 20, 20, 1, lambda i, x: x),
+    (100, 20, 20, 1, lambda i, x: map_to_test_value(x)),
+    (100000, 1000, 50000, 10, lambda i, x: x),
```

Comment:
Maybe it would be good to test behavior for some unexpected parameters, like batch_size > input_size....

Reply:
yes, it probably will fail, because I don't have safeguards in place at the moment